functions/api/v1/apps.ts

```diff
@@ -23,7 +23,24 @@ export async function handleRequest(db: D1Database, request: Request, vapidKeys:
 		return new Response('', { status: 400 })
 	}
 
-	const body = await readBody<AppsPost>(request)
+	const body: AppsPost = await readBody<AppsPost>(request)
+
+  // Parameter validation according to https://github.com/mastodon/mastodon/blob/main/app/lib/application_extension.rb
+  if ((body.client_name === undefined) || (body.client_name?.trim() === '')) {
+    return new Response('Unprocessable entity: client_name cannot be an empty string', { status: 422 })
```

Comment:
Can you return the errors as JSON?  You can reuse the logic in `backend/src/errors/index.ts`